test_conformance/api/test_clone_kernel.cpp

```diff
@@ -233,16 +236,128 @@ int test_double_arg_clone(cl_device_id deviceID, cl_context context, cl_command_
     error = clEnqueueReadBuffer(queue, bufOut, CL_TRUE, 0, BUF_SIZE, pbufRes, 0, NULL, NULL);
     test_error( error, "clEnqueueReadBuffer failed." );
 
-    if (abs(((cl_double*)pbufRes)[2] - d) > 0.0000001)
+    test_assert_error(abs(((cl_double*)pbufRes)[2] - d) <= 0.0000001,
+                      "clCloneKernel test failed.");
+
+    return TEST_PASS;
+}
+
+int test_args_enqueue_helper(cl_context context, cl_command_queue queue,
+                             cl_kernel* srcKernel, cl_int* value,
+                             cl_mem* bufOut)
```

Comment:
Minor: I think we should pass the kernel, the value, and the buffer by value, rather than as a pointer.